### DISCUSSION TITLE
Getting dependencies from off-CRAN

### DISCUSSION THREAD

**Discussion by barryrowlingson** [r-universe-org/help#154] [Q&A] [answered]
I'm trying to use R-Universe to build package B, which depends on package A. Package A has been archived from CRAN (so B has fallen off too) and some Mac people want to use B, and I don't have a Mac and they are having errors with "xcode" and "libgfortran" and stuff I know nothing about and they've tried but failed to fix.

I've got package A compiled and the user says it works. But now B depends on A, so I suspect the R-universe build will fail because A is not on CRAN. Will find out when the next hourly job runs. Is there a way I can add my universe repo to the build so it finds A okay?

**Comment by jeroen** [comment, marked as answer]
You can either explicitly add the upstream git url of package A to your personal `packages.json`, or you can specify it as a `Remote:` in the description of package B. Either has the same effect, that package A will get included in your repository, such that it is no longer depending on CRAN.

**Reply by barryrowlingson** [reply]
Ah that magic has worked because A is in my packages.json, and hence in my repo. I guess this is the build process listing the repositories:

![image](https://user-images.githubusercontent.com/888980/167444718-61235e57-3092-4625-9e80-caa615a02476.png)

Nice.
